### ISSUE TITLE
Accounts running C# script crashing at autosave

### ISSUE THREAD

**Issue by 5tanly** [ORelio/Minecraft-Console-Client#994]
https://github.com/5tanly/JoinRealm/tree/testing

I'm trying to create a script that will send messages in a loop to try and join a realm on the server. The script is triggered on first login with the tasks.ini file and seems to work fine up until the server does its autosaving plugin.
I think while the server is autosaving it causes some lag and the accounts timeout.
I believe it has something to do with the while loop. I tried to make it so when the account is already in the realm the loop would be destroyed so there's no extra processes running except for the one that tries to match the message "You have teleported to Hub Realm" in order to restart the looping (ex when the realm reboots, the server kicks you to the hub)

I am using build 275 I believe so I will update to a later version but I just posted this to see if there's an issue with my code or an issue with MCC code that would require looking at. 

Mono version is 6.8.0.96 (docker:latest)
Running on a Debian 10
This has been confirmed issue on windows aswell

**Comment by ORelio** [comment]
Your bot looks nice. I'd suggest to override the `Update()` method instead of using threads because MCC might not be fully thread safe. Also, when using Mono, there is a silent disconnect issue (#879) that may arise, we are still unsure of the root cause of this issue. If your client stops working without printing a message, it may be caused by this issue.

**Comment by 5tanly** [comment]
I saw the Update method before I started this and was worried that sending messages every 100ms would kick me for spam so I skipped over that, until I saw your [sample-script-with-world-access.cs](https://github.com/ORelio/Minecraft-Console-Client/blob/ecf0114f626ac2ee6ae8ffbf2cceb69b7e0b1b09/MinecraftClient/config/sample-script-with-world-access.cs#L40) I implemented `Update()` and it seems to be working fine now, so I will close the issue.

I might reopen in 12 hours if they crash; the reason is that I have noticed if they get kicked to the hub once, after previously joining a realm, and then rejoining the realm they will get disconnected when it autosaves. They seem to be working fine for the first autosaves, but when they rejoin after the realm reboots thats when it happens.

Also about #879, I was told by someone testing this with me that it also happens on their windows computer. I'm trying to get the details of what OS and build number they are using and will update if I get it.

**Comment by 5tanly** [comment]
I changed my script in [this commit](https://github.com/5tanly/JoinRealm/commit/689b1346dfc70dabd55caece5fd990b05bfb86bd#diff-f38aa7eb339d3c1390391139e9b2acfd) and shouldn't be using any of the threading code from before. I updated to the latest development build 285. Even then its still crashing and its really looking like #879. I have not tested this on windows yet so I'm unable to confirm if it's something in my script or that silent disconnect from mono.
